### PR TITLE
Prometheus config generation

### PR DESCRIPTION
**Pull request by Praveenrajmani**
Support for Prometheus jwt token generation

Will have admin API's for jwt generation - `mc admin prometheus --generate myminio`

### CODE REVIEW

**Comment by Praveenrajmani** [comment]
I will send another PR after this for listing metrics, say `mc admin prometheus myminio --json`

**Comment by Praveenrajmani** [comment]
Can i get some reviews @donatello @harshavardhana

File: docs/minio-admin-complete-guide.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ policy       manage canned policies
 config       manage configuration file
 heal         heal disks, buckets and objects on MinIO server
 top          provide top like statistics for MinIO
+prometheus   provide prometheus jwt token
```
Comment (harshavardhana) [inline docs/minio-admin-complete-guide.md]: This is an incorrect description, we should say that `mc admin prometheus` is to manage prometheus metrics on server

Comment (harshavardhana) [reply docs/minio-admin-complete-guide.md]: This jwt is only a sub command we are going to have more changes here. Such as bring this as a way to enable disable metrics etc. Change the prometheua auth style on server etc.

Comment (Praveenrajmani) [reply docs/minio-admin-complete-guide.md]: Done :+1:

**Comment by Praveenrajmani** [comment]
@abperiasamy approved this CLI command

**Comment by Praveenrajmani** [comment]
The idea is to have a command like `mc admin prometheus --generate` can generate the prometheus config file itself  (yaml) with all the defaults and with `bearer-token`. And `mc admin prometheus` can have sub-commands later

**Comment by Praveenrajmani** [comment]
`mc admin prometheus --generate <alias>` will generate a prometheus config which can be pasted in `prometheus.yml` - by replacing the `scrape_configs` section while starting the prometheus monitoring server

**Comment by Praveenrajmani** [comment]
`prometheus` can have sub-commands on metrics-querying which will be discussed after this

**Comment by Praveenrajmani** [comment]
PTAL @harshavardhana @donatello

**Comment by Praveenrajmani** [comment]
PTAL @harshavardhana @donatello

**Comment by Praveenrajmani** [comment]
PTAL @harshavardhana @donatello

File: docs/minio-admin-complete-guide.md [hunk 2]
```diff
@@ -221,6 +222,7 @@ Skip SSL certificate verification.
 |[**config** - manage server configuration file](#config)|
 |[**heal** - heal disks, buckets and objects on MinIO server](#heal) |
 |[**top** - provide top like statistics for MinIO](#top) |
+|[**prometheus** - manages prometheus settings](#prometheus) |
```
Comment (kannappanr) [inline docs/minio-admin-complete-guide.md]:
```suggestion
|[**prometheus** - provides prometheus config settings](#prometheus) |
```

File: docs/minio-admin-complete-guide.md [hunk 3]
```diff
@@ -499,3 +501,28 @@ mc admin trace myminio
...
 ...
 ```
+
+<a name="prometheus"></a>
+### Command `prometheus` - Manages prometheus settings
+`--generate` flag generates the prometheus config (To be pasted in `prometheus.yml`)
+
+``` sh
+NAME:
+    mc admin prometheus - provide prometheus jwt token
```
Comment (kannappanr) [inline docs/minio-admin-complete-guide.md]:
```suggestion
    mc admin prometheus - provides prometheus config
```

File: cmd/admin-prometheus.go [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+	return nil
+}
+
+// mainAdminPrometheus is the handle for "mc admin prometheus" command.
+func mainAdminPrometheus(ctx *cli.Context) error {
+
+	console.SetColor("yaml", color.New(color.FgGreen))
+
+	isGenerate := ctx.Bool("generate")
+	if !isGenerate {
```
Comment (donatello) [inline cmd/admin-prometheus.go]: If the flag is always required it is never actually useful. The command can just be `mc admin prometheus myminio`, consistent with other commands.

**Comment by donatello** [comment]
Ah noticing the previous discussion, the intention seems to be that more subcommands would be added later. Then it should be `mc admin prometheus generate myminio`. This makes generate a subcommand rather than a compulsory option.

**Comment by Praveenrajmani** [comment]
PTAL @donatello @harshavardhana @kannappanr

**Comment by kannappanr** [comment]
@Praveenrajmani Please resolve the conflicts.

**Comment by Praveenrajmani** [comment]
Resolved the conflicts :+1:  @kannappanr

**Comment by Praveenrajmani** [comment]
@kannappanr This PR should be blocked until these two PRs gets merged

- https://github.com/minio/cookbook/pull/252
- https://github.com/minio/minio/pull/7640

**Comment by kannappanr** [comment]
@Praveenrajmani We can merge this PR first. `mc` is not talking to `MinIO` server in this PR, so it is fine to take this PR in.

**Pull request merged**